wpi-many.sh

```diff
@@ -0,0 +1,233 @@
+    exit 2
+fi
+
+if [ ! -d "${CHECKERFRAMEWORK}" ]; then
+    echo "CHECKERFRAMEWORK is set to a non-existent directory ${CHECKERFRAMEWORK}"
+    exit 2
+fi
+
+if [ "x${OUTDIR}" = "x" ]; then
+    echo "You must specify an output directory using the -o argument."
+    exit 3
+fi
+
+if [ "x${INLIST}" = "x" ]; then
+    echo "You must specify an input file using the -i argument."
+    exit 4
+fi
+
+if [ "x${GITHUB_USER}" = "x" ]; then
+    GITHUB_USER="${USER}"
+fi
+
+export JAVA_HOME="${JAVA11_HOME}"
+
+### Script
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+    
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
+
+while IFS='' read -r line || [ "$line" ]
+do    
+    REPOHASH=${line}
+
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
+
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
+    REPO_NAME_HASH="${REPO_NAME}-${HASH}"
+
+    # Use repo name and hash, but not owner.  We want
+    # repos that are different but have the same name to be treated
+    # as different repos, but forks with the same content to be skipped.
+    # TODO: consider just using hash, to skip hard forks?
+    mkdir -p "${REPO_NAME_HASH}"
+
+    pushd "${REPO_NAME_HASH}" || exit 5
+    
+    if [ ! -d "${REPO_NAME}" ]; then
+        # The "GIT_TERMINAL_PROMPT=0" setting prevents git from prompting for
+	# username/password if the repository no longer exists.
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
+        # Skip the rest of the loop and move on to the next project
+	# if cloning isn't successful.
+        if [ ! -d "${REPO_NAME}" ]; then
+           continue
+        fi
+    else
+        rm -rf "${REPO_NAME}/dljc-out"
+    fi
+
+    pushd "${REPO_NAME}" || exit 5
+
+    git checkout "${HASH}"
+
+    OWNER=$(echo "${REPO}" | cut -d / -f 4)
+
+    if [ "${OWNER}" = "${GITHUB_USER}" ]; then
+        ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
+        git remote add unannotated "${ORIGIN}"
+    fi
+
+    REPO_FULLPATH=$(pwd)
+    
+    popd || exit 5
+
+    RESULT_LOG="${OUTDIR}-results/${REPO_NAME_HASH}-wpi.log"
+    touch "${RESULT_LOG}"
+
+    "${SCRIPTDIR}/wpi.sh" -d "${REPO_FULLPATH}" -u "${GITHUB_USER}" -t "${TIMEOUT}" -- "$@" &> "${RESULT_LOG}"
+
+    popd || exit 5
+
+    # If the result is unusable (i.e. wpi cannot run),
+    # we don't need it for data analysis and we can
+    # delete it right away.
+    if [ -f "${REPO_FULLPATH}/.cannot-run-wpi" ]; then
+        rm -rf "${REPO_NAME_HASH}" &
+    else
+        cat "${REPO_FULLPATH}/dljc-out/wpi.log" >> "${RESULT_LOG}"
+    fi
+
+    cd "${OUTDIR}" || exit 5
+    
+done <"${INLIST}"
+
+popd || exit 5
+
+## This section is here rather than in wpi-summary.sh because cloc can be moderately expensive.
```

Comment:
I get the impression that cloc has been largely abandoned by users because of performance issues.
You might want to switch to a faster tool such as https://github.com/boyter/scc .
It feels instantaneous even on large projects.
As side benefits, you can remove the dependency on perl and can better organize the scripts.